app/src/main/java/zapsolutions/zap/fragments/OpenChannelBSDFragment.java

```diff
@@ -43,24 +79,185 @@ public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup c
+        mBtnNumpad[2] = view.findViewById(R.id.Numpad3);
+        mBtnNumpad[3] = view.findViewById(R.id.Numpad4);
+        mBtnNumpad[4] = view.findViewById(R.id.Numpad5);
+        mBtnNumpad[5] = view.findViewById(R.id.Numpad6);
+        mBtnNumpad[6] = view.findViewById(R.id.Numpad7);
+        mBtnNumpad[7] = view.findViewById(R.id.Numpad8);
+        mBtnNumpad[8] = view.findViewById(R.id.Numpad9);
+        mBtnNumpad[9] = view.findViewById(R.id.Numpad0);
+
+        mBtnNumpadDot = view.findViewById(R.id.NumpadDot);
+        mBtnNumpadBack = view.findViewById(R.id.NumpadBack);
+
+        // Set action for numpad number buttons
+        for (Button btn : mBtnNumpad) {
+            btn.setOnClickListener(v -> {
+                // Add input
+                int start = Math.max(mEtAmount.getSelectionStart(), 0);
+                int end = Math.max(mEtAmount.getSelectionEnd(), 0);
+                mEtAmount.getText().replace(Math.min(start, end), Math.max(start, end),
+                        btn.getText(), 0, btn.getText().length());
+
+            });
+        }
+
+        mOkButton.setOnClickListener(v -> dismiss());
+
+        // Action when clicked on "x" (close) button
+        ImageButton btnCloseBSD = view.findViewById(R.id.closeButton);
+        btnCloseBSD.setOnClickListener(v -> dismiss());
+
+        // Set action for numpad "." button
+        mBtnNumpadDot.setOnClickListener(v -> {
+            // Add input
+            int start = Math.max(mEtAmount.getSelectionStart(), 0);
+            int end = Math.max(mEtAmount.getSelectionEnd(), 0);
+            mEtAmount.getText().replace(Math.min(start, end), Math.max(start, end),
+                    mBtnNumpadDot.getText(), 0, mBtnNumpadDot.getText().length());
+        });
+
+        // Set action for numpad "delete" button
+        mBtnNumpadBack.setOnClickListener(v -> {
+            // remove Input
+            deleteAmountInput();
+        });
+
+        setAvailableFunds();
+
+        // Input validation for the amount field.
+        mEtAmount.addTextChangedListener(new TextWatcher() {
+
+            @Override
+            public void beforeTextChanged(CharSequence arg0, int arg1,
+                                          int arg2, int arg3) {
+                // TODO Auto-generated method stub
 
-        // temporary coming soon
-        mOpenChannelButton.setText(R.string.coming_soon);
-        mLocalAmountsInputView.setEnabled(false);
+            }
+
+            @Override
+            public void afterTextChanged(Editable arg0) {
+                // remove the last inputted character if not valid
+                if (!mAmountValid) {
+                    deleteAmountInput();
+                }
+
+                mUseValueBeforeUnitSwitch = false;
+
+                if (!mEtAmount.getText().toString().equals(".")) {
+                    // make text red if input is too large
+                    long maxSendable;
+                    long minSendable = 20000;
```

Comment:
This value is not obvious to the user.
When starting to type numbers in satoshis you cannot click open until you are higher than 20000. This might seem strange to unexperienced users.
Maybe we can show the minimum amount instead of the "Max available" WHILE the inputted number is less than the minimum.